python/GafferSceneUI/ShaderTweakProxyUI.py

```diff
@@ -298,6 +290,22 @@ def _plugContextMenu( plug, shaderTweaks ) :
 	return menuDefinition
 
 def __plugContextMenuSignal( graphEditor, plug, menuDefinition ) :
+
+	if plug.direction() != Gaffer.Plug.Direction.In:
+		return
+
+	# Find the actual node if we're looking at something like a box input
+	# NOTE : This could fail if a shader output is connected to 2 things, and the first thing is not a shader,
+	# but that seems like a pretty weird case, and we want to get to the early out without doing too much traversal
+	destPlug = Gaffer.PlugAlgo.findDestination( plug, lambda plug : plug if not plug.outputs() else None )
```

Comment:
I don't think "too much traversal" is really a worry here is it? We use `findDestination()` in far more performance-critical places, like PlugValueWidget construction. The more canonical thing seems to be to use the Shader/ShaderTweaks check as the predicate, and then we don't need this NOTE at all.